### PR TITLE
Feature/intl

### PR DESCRIPTION
**Pull request by Isaakske**
## CLA

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [X] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

An implementation of internationalization.
Using the [react-intl](https://github.com/formatjs/react-intl) library to wrap the Chat component in a Provider that allows users of the package to pass their own locale and messages props.
All text currently present in the project has been extracted, providing translation keys as well as default messages. 

This is not a true internationalization of the project, as no languages other than the default English have been provided. This implementation simply gives the package user the possibility to change, add, or translate the keys for their own purposes.

As far as step 3 of the checklist goes (testing), I'm not sure how to proceed with that. I'm under the assumption that this refers to unit tests, but I don't really know how to, or what the correct location for these tests would be. Of course this feature has been used and checked in the example projects, as well as an implementation in one of our own projects. 

I'm sorry if I missed anything, as this is my first PR for this project.

### CODE REVIEW

**Comment by Isaakske** [comment]
While this feature allows users to pass their own set of messages, there are plenty of files where moment.js is used to format dates, show the names of days, ...

The locale of moment.js can be set at runtime, however every locale requires an additional import.

Seeing as it is currently the package user's responsibility to provide messages as well as their own extra locale(s), I thought it made sense to shift the responsibility of moment.js' locale to their side as well. That way they can import their necessary set of locales for moment.js, and this project doesn't needlessly import all possible options, increasing the size.

I'm not entirely sure if this was the right approach, and am open to ideas or discussions regarding this.

[It does work.](https://i.imgur.com/6yeBmTQ.png)

**Comment by ferhatelmas** [comment]
@Isaakske This is a huge contribution, thank you very much. 

It's also a big maintenance burden. I am not sure if this feature is wanted in terms of package selection, timing, etc.

Just in case, next time, please communicate before doing any real work so that nothing is wasted.(Please ignore if it's done, I'm around recently)

As a quick look, it looks good and IMHO, your take related to packages that come with their own locales like moment sounds reasonable to me.

Thank you again for your work, please be patient. Team is busy on a few features and when they have some time, you will definitely get a good/fair review.

File: src/components/Chat.js [hunk 1]
```diff
@@ -72,10 +74,16 @@ export class Chat extends PureComponent {
...
     theme: PropTypes.string,
+    /** The App locale */
+    locale: PropTypes.string,
+    /** The messages for the given locale */
+    messages: PropTypes.object,
   };
 
   static defaultProps = {
     theme: 'messaging light',
+    locale: 'en',
```
Comment (jaapbakker88) [inline src/components/Chat.js]: @Isaakske do we still need this prop if we're not using it for setting moments locale

Comment (jaapbakker88) [reply src/components/Chat.js]: also, can you give an example of how you set Moment.js locale so we can get the date's translated as well.

**Comment by Isaakske** [comment]
Apologies, I was on a holiday and didn't get a chance to reply until now.

@ferhatelmas We needed internationalization in our implementation of stream-chat-react. I figured if I had to do it anyway I might as well keep it in a separate branch and PR it :wink: 
Next time I'll make sure to communicate with a member of the team.

File: src/components/Chat.js (hunk 1 above)
Comment (Isaakske) [reply src/components/Chat.js]: [The documentation](https://github.com/formatjs/react-intl/blob/master/docs/Getting-Started.md#creating-an-i18n-context) shows the prop being passed, even when only the appropriate set of messages is passed. I don't know the exact reasoning behind it.

Comment (Isaakske) [reply src/components/Chat.js]: A simple way to set it in the root component:

`import moment from 'moment';
import 'moment/locale/nl';

moment.locale('nl');`

**Comment by jaapbakker88** [comment]
No problem, I'll have a look over it again this week and hopefully we can merge it quickly.

**Comment by vishalnarkhede** [comment]
Hey @Isaakske ,

I have created a feature branch for internationalization - `feature/internationalization`. Can you please change the target branch to `feature/internationalization` instead of master. We will have to do doc updates and few changes on our end on top of these code changes, so can't merge directly to master.

P.S. Thanks for all the effort :)

**Comment by jaapbakker88** [comment]
@Isaakske any update on this?

**Comment by vishalnarkhede** [comment]
Going to close it now. We published i18n feature in v0.10.1!

**Pull request closed without merging**